toolchain/check/import_cpp.cpp

```diff
@@ -373,6 +374,105 @@ static auto ClangLookup(Context& context, SemIR::NameScopeId scope_id,
   return lookup;
 }
 
+// Looks up for constructors in the class scope and returns the lookup result.
+static auto ClangConstructorLookup(const Context& context,
+                                   SemIR::NameScopeId scope_id)
+    -> clang::DeclContextLookupResult {
+  const SemIR::NameScope& scope = context.sem_ir().name_scopes().Get(scope_id);
+
+  clang::Sema& sema = context.sem_ir().cpp_ast()->getSema();
+  clang::Decl* decl =
+      context.sem_ir().clang_decls().Get(scope.clang_decl_context_id()).decl;
+  return sema.LookupConstructors(clang::cast<clang::CXXRecordDecl>(decl));
+}
+
+// Returns true if the given Clang declaration is the implicit injected class
+// name within the class.
+static auto IsDeclInjectedClassName(const Context& context,
+                                    SemIR::NameScopeId scope_id,
+                                    SemIR::NameId name_id,
+                                    const clang::NamedDecl* named_decl)
+    -> bool {
+  if (!named_decl->isImplicit()) {
+    return false;
+  }
+
+  const auto* record_decl = clang::dyn_cast<clang::CXXRecordDecl>(named_decl);
+  if (!record_decl) {
+    return false;
+  }
+
+  SemIR::ClangDecl clang_decl = context.sem_ir().clang_decls().Get(
+      context.sem_ir().name_scopes().Get(scope_id).clang_decl_context_id());
+  const auto* scope_record_decl =
+      clang::cast<clang::CXXRecordDecl>(clang_decl.decl);
+
+  const clang::ASTContext& ast_context =
+      context.sem_ir().cpp_ast()->getASTContext();
+  CARBON_CHECK(
+      ast_context.getCanonicalType(
+          ast_context.getRecordType(scope_record_decl)) ==
+      ast_context.getCanonicalType(ast_context.getRecordType(record_decl)));
+
+  CARBON_CHECK(name_id ==
+               context.sem_ir()
+                   .classes()
+                   .Get(context.sem_ir()
+                            .insts()
+                            .GetAs<SemIR::ClassDecl>(clang_decl.inst_id)
+                            .class_id)
+                   .name_id);
+  return true;
+}
+
+// Look ups the given name in the Clang AST in a specific scope. If the found
+// declaration is the injected class name, look ups constructors instead. If not
+// found, returns `nullopt`. If there's not a single result, returns `nullptr`.
+// Otherwise, returns the single declaration.
+static auto ClangLookup(Context& context, SemIR::LocId loc_id,
+                        SemIR::NameScopeId scope_id, SemIR::NameId name_id)
+    -> std::optional<clang::NamedDecl*> {
+  auto lookup = ClangLookupName(context, scope_id, name_id);
+  if (lookup) {
+    if (!lookup->isSingleResult()) {
+      context.TODO(loc_id,
+                   llvm::formatv("Unsupported: Lookup succeeded but couldn't "
+                                 "find a single result; LookupResultKind: {0}",
+                                 static_cast<int>(lookup->getResultKind()))
+                       .str());
```

Comment:
```suggestion
                                 static_cast<int>(lookup->getResultKind())));
```

I think this `str` isn't needed here

Reply:
Done.

Reply:
Did this change get dropped? I'm still seeing it in the latest commit

Reply:
Done, not sure what happened.
Sorry about that.